packages/main/src/plugin/provider-registry.ts

```diff
@@ -1325,4 +1325,13 @@ export class ProviderRegistry {
       throw error;
     }
   }
+
+  getProviderInfo(internalProviderId: string): ProviderInfo | undefined {
+    try {
+      const provider = this.getMatchingProvider(internalProviderId);
+      return this.toProviderInfo(provider);
+    } catch (e) {
+      return undefined;
+    }
+  }
```

Comment:
issue: I'm not sure we should use getMatchingProvider because it's throwing an error that we need to catch here
`this.providers.get(internalId)` is returning directly the provider implementation


```suggestion
    const provider = this.providers.get(id);
    if (provider) {
      return this.toProviderInfo(provider);
    }
    return undefined;
```